### PR TITLE
Improving documentation

### PR DESCRIPTION
**Pull request by ant0nsc**
* Cleaning up examples, adding highlighting of lines
* Onboarding section
* Fix rendering issues for bullet points
* Ensuring that class and function doc render correctly

### CODE REVIEW

File: docs/source/first_steps.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# First steps: How to get started with hi-ml
+
+## Setting up AzureML
+You need to have an AzureML workspace in your Azure subscription.
+Download the config file from your AzureML workspace, as described 
+[here](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-configure-environment). **Put this file (it
```
Comment (mebristo) [inline docs/source/first_steps.md:6]: nit: unmatched bracket

File: docs/source/index.rst [hunk 2]
```diff
@@ -3,14 +3,19 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
-Welcome to hi-ml's documentation!
+Documentation for the Health Intelligence Machine Learning toolbox hi-ml
 =================================
```
Comment (mebristo) [inline docs/source/index.rst:7]: I think this might cause an error in preview since the underline needs to be at least the same length as the title

Comment (ant0nsc) [reply docs/source/index.rst:7]: Thanks, fixed! That type of requirements is why I prefer MD over RST...

File: src/health/azure/himl_download.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from health.azure.azure_util import AzureRunIdSource, get_aml_runs
 
 from health.azure.himl import get_workspace
-from health.azure.run_tensorboard import determine_run_id_source
+from health.azure.himl_tensorboard import determine_run_id_source
```
[the remark this answers is not in the document]
Comment (ant0nsc) [reply src/health/azure/himl_download.py:12]: Yes, I'd leave that out of there. I have a couple of other ideas with those scripts.

**Review by mebristo — APPROVED** [review]

**Pull request merged**